Lib/test/test_venv.py

```diff
@@ -872,6 +872,24 @@ def test_venv_same_path(self):
                 else:
                     self.assertFalse(same_path(path1, path2))
 
+    # gh-126084: venvwlauncher should run pythonw, not python
+    @requireVenvCreate
+    @unittest.skipUnless(os.name == 'nt', 'only relevant on Windows')
+    def test_venvwlauncher(self):
+        """
+        Test that the GUI launcher runs the GUI python.
+        """
+        rmtree(self.env_dir)
+        venv.create(self.env_dir)
+        exename = "pythonw_d.exe"
```

Comment:
Will this work on a non-debug build?

Reply:
Just fixed that in the previous commit. What it won't work on is a free-threaded build, though.

Reply:
That was the original problem, actually; the GitHub CI environment _is_ a debug build (as should have been obvious), and that is why the first few attempts failed.